projects/kit/components/input-date-time/input-date-time.component.ts

```diff
@@ -143,24 +143,19 @@ export class TuiInputDateTimeComponent
     }
 
     get computedValue(): string {
-        const {value, nativeValue, focused, touched} = this;
+        const {value, nativeValue, timeMode} = this;
         const [date, time] = value;
+        const hasTimeInputChars = nativeValue.length > this.dateFiller.length;
 
-        if (
-            (date && !nativeValue) ||
-            (date && nativeValue.length === this.dateFiller.length) ||
-            (date && time)
-        ) {
-            return `${date.toString()}${DATE_TIME_SEPARATOR}${
-                time ? time.toString(this.timeMode) : ''
-            }`;
+        if (!date) {
+            return nativeValue;
         }
 
-        if (touched || focused) {
+        if (!time && hasTimeInputChars) {
```

Comment:
nit: don't you want to combine two conditionals returned nativeValue in one?

Reply:
done